eZ/Publish/Core/MVC/Symfony/SiteAccessGroup.php

```diff
@@ -8,7 +8,9 @@
 
 namespace eZ\Publish\Core\MVC\Symfony;
 
-final class SiteAccessGroup
+use JsonSerializable;
+
+final class SiteAccessGroup implements JsonSerializable
```

Comment:
Both locations that perform `SiteAccess` serialization for the purposes of storing them in `Request` attributes object have access to the serializer service. I think that using it is preferable over maintaining encoding (in this file) and decoding (in the listener).

Reply:
sorry for late replay, 
true, but:
If someone is relying on `JsonSerializable` and current state of things, he would get wrongly serialized object. So that's need to be added/fixed as well, and as a followup/additional step it could be redone to use Serializer, right?

Reply:
AFAIK n this case we always  rely on native PHP serialization nad later on symfony/serializer :wink:

Reply:
> If someone is relying on JsonSerializable and current state of things

Currently calling `json_encode` on this object is an error (if not straight up PHP error, it is a logical error - this object was not supposed to be json encoded this way).

Adding this interface is only acceptable if we are treating this change as internal only, and only to make it slightly faster to implement. However, I would say that a proper normalizer should be added instead, as the object should not be aware how it is presented in REST (I assume that's the target).